packages/wonder-blocks-form/src/components/__tests__/labeled-text-field.test.js

```diff
@@ -0,0 +1,50 @@
+//@flow
+import * as React from "react";
+import {mount} from "enzyme";
+
+import LabeledTextField from "../labeled-text-field.js";
+
+const wait = (delay: number = 0) =>
+    new Promise((resolve, reject) => {
+        // eslint-disable-next-line no-restricted-syntax
+        return setTimeout(resolve, delay);
+    });
+
+describe("LabeledTextField", () => {
+    it("labeledtextfield becomes focused", () => {
+        // Arrange
+        const wrapper = mount(<LabeledTextField />);
+        const field = wrapper.find("TextField");
+
+        // Act
+        field.simulate("focus");
+
+        // Assert
+        expect(wrapper).toHaveState("focused", true);
+    });
```

Comment:
It might be worth adding a snapshot test for this state.

Reply:
How do you suggest I add this snapshot? Would it be in the `custom-snapshot.test.js` file? If so, how would I change the `focused` state since it is **not** a prop state?

Reply:
I was going to suggest doing something similar to what's in https://github.com/Khan/wonder-blocks/blob/75334ca695dc97adcc0c557c2530b3dc95af8925/packages/wonder-blocks-core/src/components/__tests__/view.test.js, but I see now that none of those tests are simulating events.  I think you might be to trigger an event on the object returned by `renderer.create(node)` but since this isn't something we already please ignore.  A lot of our other components have a stateless (*-core.js) and stateful halves so testing the markup of the focus state can be done by taking a snapshot of the stateless component with the right props.  Please ignore my suggestion for now.  In the future this kind of testing might get easy if we adopt React Testing Library.

Reply:
Alright. Thanks for the info! It seems React Testing Library will be giving us so much more flexibility in many domains, can't wait to [hopefully] see it soon.